### PR TITLE
fix: add support for numpy scaler types in type_engine

### PR DESCRIPTION
**Pull request by techytushar**
# TL;DR
Fixes: https://github.com/flyteorg/flyte/issues/2753

Added support for numpy scaler types in the `type_engine.py` file

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

* Updated the `SimpleTransformer` class to take tuples of types instead of single type, so `int` transformer now supports both `int` and `numpy.integer`
* Also updated the type checking methods to use `isInstance` instead of exact type matches

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2753

## Follow-up issue
_NA_

### CODE REVIEW

**Review by pingsutw — COMMENTED** [review]
@techytushar Thanks for your contribution. Could you fix the test error and add a test to [test_type_engine.py](https://github.com/flyteorg/flytekit/blob/master/tests/flytekit/unit/core/test_type_engine.py)?

**Comment by techytushar** [comment]
Accidently closed this PR, have opened a new one with added tests https://github.com/flyteorg/flytekit/pull/1258
